moveit_core/planning_interface/include/moveit/planning_interface/planning_request.h

```diff
@@ -36,12 +36,22 @@
 
 #pragma once
 
+#include <Eigen/Core>
 #include <moveit_msgs/msg/motion_plan_request.hpp>
 
 namespace planning_interface
 {
-// for now this is just a typedef
 
-typedef moveit_msgs::msg::MotionPlanRequest MotionPlanRequest;
+/** \brief Definition for a cost function to measure the cost of a single state during motion planning */
+using StateCostFn = std::function<double(const Eigen::VectorXd& state_vector)>;
+
+struct MotionPlanRequest : moveit_msgs::msg::MotionPlanRequest
+{
+  MotionPlanRequest(moveit_msgs::msg::MotionPlanRequest request_msg = moveit_msgs::msg::MotionPlanRequest(),
+                    planning_interface::StateCostFn state_cost_function = nullptr);
+  planning_interface::StateCostFn state_cost_function = nullptr;
+
+  [[nodiscard]] moveit_msgs::msg::MotionPlanRequest toMessage() const;
```

Comment:
I'm confused about why this is needed. Since this is inheriting from `moveit_msgs::msg::MotionPlanRequest` and we are not doing anything special with the `state_cost_function` member, this shouldn't be needed, right?

Reply:
I don't remember if it was at runtime or during compilation but in either one of these situations the state_cost_function member function pointer could not be handled and I got either a compiler error or a segfault

Reply:
hmm... maybe worth investigating more? See my other comment about the `reinterpret_cast` to the base class. Or I believe a `static_cast` would work too, since you'd be casting to the base class.